### PR TITLE
petsc: fix pkgconfig error - with incorrect library link order that c…

### PR DESCRIPTION
**Pull request by balay**
…an break mfem examples with static build

cc: @keitat

### CODE REVIEW

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ class Petsc(Package):
         patch('macos-clang-8.1.0.diff',
               when='@3.7.5%clang@8.1.0:')
 
+    patch('petsc-pkgconfig.patch', when='@3.8.0:3.8.1')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/petsc/package.py]: Is there way to pull it from bitbucket.org? (i.e. https://bitbucket.org/petsc/petsc/commits/SHA/raw)

Comment (balay) [reply var/spack/repos/builtin/packages/petsc/package.py]: `spack fetch` does not download url patches?  I need that to work.

If I can't add patch to repo - I guess I'll have to spin a 3.8.2 tarball.

```

balay@asterix /home/balay/git-repo/github/spack (xsdk-0.3.0 *=)
$ git diff
diff --git a/var/spack/repos/builtin/packages/petsc/package.py b/var/spack/repos/builtin/packages/petsc/package.py
index 1262c47e3..7b2d79f7c 100644
--- a/var/spack/repos/builtin/packages/petsc/package.py
+++ b/var/spack/repos/builtin/packages/petsc/package.py
@@ -93,7 +93,9 @@ class Petsc(Package):
         patch('macos-clang-8.1.0.diff',
               when='@3.7.5%clang@8.1.0:')
 
-    patch('petsc-pkgconfig.patch', when='@3.8.0:3.8.1')
+    patch('https://bitbucket.org/petsc/petsc/commits/f5eb75d/raw',
+          sha256='b3fd34f274782a3bf4ae5a4c4cf32c291dc9448fd938a235d13b2ab3a3934262',
+          when='@3.8.0:3.8.1')
 
     # Virtual dependencies
     # Git repository needs sowing to build Fortran interface
balay@asterix /home/balay/git-repo/github/spack (xsdk-0.3.0 *=)
$ spack fetch petsc
==> Fetching http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-3.8.1.tar.gz
######################################################################## 100.0%
balay@asterix /home/balay/git-repo/github/spack (xsdk-0.3.0 *=)
$ find . -type f -name petsc*.tar.gz
./var/spack/cache/petsc/petsc-3.8.1.tar.gz
balay@asterix /home/balay/git-repo/github/spack (xsdk-0.3.0 *=)
$ ls var/spack/cache/petsc/
petsc-3.8.1.tar.gz
balay@asterix /home/balay/git-repo/github/spack (xsdk-0.3.0 *=)
$ 
```

Comment (junghans) [reply var/spack/repos/builtin/packages/petsc/package.py]: I see! Is that #5344?

Comment (balay) [reply var/spack/repos/builtin/packages/petsc/package.py]: I'll just spin 3.8.2 - to get this issue resolved for now

Comment (junghans) [reply var/spack/repos/builtin/packages/petsc/package.py]: I can merge it as well...

Comment (balay) [reply var/spack/repos/builtin/packages/petsc/package.py]: Already have 3.8.2

**Comment by balay** [comment]
Closing in favor of https://github.com/spack/spack/pull/6244

**Pull request closed without merging**